src/viam_mlmodelservice_triton_impl.cpp

```diff
@@ -338,6 +340,90 @@ class Service : public vsdk::MLModelService, public vsdk::Stoppable, public vsdk
         return state_;
     }
 
+    static std::string initialize_directory_(std::string model_name, std::int64_t version, bool is_tf) {
+        const char* base_directory = std::getenv("VIAM_MODULE_DATA");
+
+        std::stringstream ss;
+        ss << base_directory << "/" << model_name;
+        if (is_tf) {
+            // TensorFlow models need one extra directory to work with
+            // Triton. So, the symlink to the directory containing the
+            // data will be named "model.savedmodel" inside the version
+            // directory, rather than being the version directory
+            // itself.
+            ss << "/" << version;
+        }
+        const std::string directory_name = ss.str();
+
+        std::error_code ec;
+        bool success = std::filesystem::create_directories(directory_name, ec);
+        // If the directory already exists, `success` will be false but the
+        // error code will be 0. Don't treat that as an error.
+        if (!success && ec.value() != 0) {
+            std::ostringstream buffer;
+            buffer << service_name << ": cannot create directory structure";
+            throw std::runtime_error(buffer.str());
+        }
```

Comment:
Why use the `error_code` taking form just to convert it to an exception? I think you could just call `std::filesystem::create_directories(directory_name)` and it'll throw `filesystem_error` on error, which is already a `std::runtime_error`.